File: src/helpText.rs
// helpContent.ts
// Centralized help text for the ABM interface, to be injected into your UI via innerHTML or as a React component.

export const helpContent = {
  general: `
    <h2>Agent-Based Inheritance & Socioeconomic Simulator</h2>
    <p>This interactive model shows how traits and resources flow from one generation to the next through four stages:</p>
    <ol>
      <li><strong>Parent Genes → Genes</strong></li>
      <li><strong>Parent Wealth → Environmental Endowment</strong></li>
      <li><strong>Genes + Environment → Education Level</strong></li>
      <li><strong>Education + Parent Wealth → Child Wealth</strong></li>
    </ol>
    <p>Each simulated “agent” inherits genetic values and an environmental endowment, which together determine its educational attainment. That education—combined with parental wealth—then produces the agent’s own wealth. Use the controls on the left to adjust noise, homophily, population size, etc., and watch the effects ripple through the grid and across your inequality metrics on the right.</p>
  `,
  plots: `
    <h3>Plots</h3>
    <ul>
      <li><strong>Population Raster</strong>
        <ul>
          <li>Shows every agent as a cell, colored by whichever stage you’ve selected: Genes, Environmental Endowment, Education Level, or Wealth.</li>
          <li>Grid layout makes spatial patterns easy to spot.</li>
        </ul>
      </li>
      <li><strong>Lorenz Curve (Wealth Distribution)</strong>
        <ul>
          <li>Cumulative share of agents vs. cumulative share of <strong>wealth</strong>.</li>
          <li>Diagonal = perfect equality; bowing indicates inequality.</li>
        </ul>
      </li>
      <li><strong>Gini Coefficient (Wealth Inequality)</strong>
        <ul>
          <li>Single value (0–1) summarizing how unequal the wealth distribution is.</li>
        </ul>
      </li>
      <li><strong>Histogram of Selected Feature</strong>
        <ul>
          <li>Distribution of genes, environment, education, or wealth across agents.</li>
          <li>Hover bars to see exact counts or percentages.</li>
        </ul>
      </li>
      <li><strong>Time Series (Over Generations)</strong> (if enabled)
        <ul>
          <li>Tracks summary stats (mean, variance, Gini) of <strong>wealth</strong> or <strong>education</strong> as you simulate multiple generations.</li>
        </ul>
      </li>
    </ul>
  `,
  controls: `
    <h3>Controls</h3>
    <dl>
      <dt>Feature Selector</dt>
      <dd>Pick which stage to display/analyze:
        <ul>
          <li>Genes</li>
          <li>Environmental Endowment</li>
          <li>Education Level</li>
          <li>Wealth</li>
        </ul>
      </dd>
      <dt>Noise Level</dt>
      <dd>Adjust randomness in gene transmission each generation (low = near-deterministic, high = more mutation).</dd>
      <dt>Homophily</dt>
      <dd>How strongly agents choose partners with similar education or wealth (0 = random; 1 = strict; &gt;1 amplifies similarity effects).</dd>
      <dt>Population Size</dt>
      <dd>Total agents simulated (more &rarr; smoother stats, slower compute).</dd>
      <dt>Grid Dimensions</dt>
      <dd>Number of rows (columns scale accordingly) for the raster layout.</dd>
      <dt>Generations</dt>
      <dd>Number of breeding cycles to run before showing final outcomes.</dd>
      <dt>Run / Reset</dt>
      <dd>
        <ul>
          <li><strong>Run</strong>: simulates with current settings.</li>
          <li><strong>Reset</strong>: restores defaults and clears results.</li>
        </ul>
      </dd>
      <dt>Download Data</dt>
      <dd>Export the final dataset or summary statistics as CSV.</dd>
    </dl>
  `,
  extras: `
    <h3>Extras & Tooltips</h3>
    <ul>
      <li>Hover over any plot element to see precise values.</li>
      <li>Legend beneath the raster maps colors to numeric ranges.</li>
      <li>Help (<code>i</code>) icon opens a quick primer on inheritance and inequality concepts.</li>
      <li>Auto-update toggle (optional) re-runs the sim live as you drag sliders.</li>
    </ul>
  `
};

File: src/helpText.ts
// helpContent.ts
// Defines structured help sections for the ABM interface, displayed via dropdown

export interface HelpSection {
  id: string;
  title: string;
  content: string;
}

export const helpSections: HelpSection[] = [
  {
    id: 'overview',
    title: 'Overview',
    content: `
      <h2>Agent-Based Inheritance & Socioeconomic Simulator</h2>
      <p>This interactive model shows how traits and resources flow from one generation to the next through four stages:</p>
      <ol>
        <li><strong>Parent Genes → Genes</strong></li>
        <li><strong>Parent Wealth → Environmental Endowment</strong></li>
        <li><strong>Genes + Environment → Education Level</strong></li>
        <li><strong>Education + Parent Wealth → Child Wealth</strong></li>
      </ol>
      <p>Adjust the sliders to explore how inheritance, environment, education, and chance combine to shape inequality over time.</p>
    `
  },
  // {
  //   id: 'technical',
  //   title: 'Technical Architecture',
  //   content: `
  //     <h3>Technical Architecture</h3>
  //     <p>The <code>index.html</code> file defines static UI elements (divisions, canvas, controls). When you load the page, the browser parses this HTML, then executes the compiled JavaScript (from <code>main.ts</code>).</p>
  //     <p>TypeScript is a typed superset of JavaScript: you write <code>*.ts</code> files, and the <code>vite</code> tool compiles them to <code>*.js</code> for the browser. This provides strong typing and a modern development experience with zero runtime penalty.</p>
  //     <p><code>main.ts</code> initializes UI controls, injects help sections, and calls functions from <code>model.ts</code> to run the agent-based simulation each generation, including custom mating logic.</p>
  //   `
  // },
  {
    id: 'model',
    title: 'Model Structure',
    content: `
      <h3>How the Model Runs</h3>
      <ul>
        <li>Genes: each agent inherits one randomly chosen "education" gene from each parent, with occasional mutation.</li>
        <li>Environment: set by parents' wealth each generation, with optional noise added in to account for chance.</li>
        <li>Education: education success is a function of the average gene value, and the average environment value, weighted by the proportion contribution slider.</li>
        <li>Wealth: weighted mix of education vs. parental wealth (Education → Wealth slider).</li>
        <li>Mating Logic: random or assortative by genes/environment controlled by homophily sliders.</li>
      </ul>
    `
  },
  {
    id: 'plots',
    title: 'Plots',
    content: `
      <h3>Plots</h3>
      <ul>
        <li><strong>Population Raster</strong>: grid of agents colored by the selected attribute.</li>
        <li><strong>Lorenz Curve</strong>: cumulative share of agents vs. cumulative share of wealth.</li>
        <li><strong>Gini Coefficient</strong>: summary statistic (0–1) of wealth inequality.</li>
        <li><strong>Histogram</strong>: distribution of genes, environment, education, or wealth.</li>
        <li><strong>Time Series</strong>: tracks summary stats across generations.</li>
      </ul>
    `
  },
  {
    id: 'controls',
    title: 'Controls',
    content: `
      <h3>Controls</h3>
      <ul>
        <li><strong>Start</strong>: begin continuous simulation.</li>
        <li><strong>Pause</strong>: pause the simulation.</li>
        <li><strong>Reset</strong>: return to Generation 0.</li>
        <li><strong>Step</strong>: advance one generation.</li>
      </ul>
      <dl>
        <dt>Color by</dt>
        <dd>Choose Genes, Environmental Endowment, Education Level, or Wealth.</dd>
        <dt>Env Quality Noise σ</dt>
        <dd>Gaussian noise in environment (e.g., random exposures).</dd>
        <dt>Gene → Education Weight</dt>
        <dd>Mix of innate talent vs. environment in education:</dd>
        <dd>
          <ul>
            <li>1 = pure meritocratic utopia (talent only)</li>
            <li>0 = environmental determinism (parental influence only)</li>
          </ul>
        </dd>
        <dt>Education → Wealth Weight</dt>
        <dd>Mix of education vs. inheritance in wealth:</dd>
        <dd>
          <ul>
            <li>1 = meritocracy (education only)</li>
            <li>0 = aristocracy (inheritance only)</li>
          </ul>
        </dd>
        <dt>Finance Success Noise σ</dt>
        <dd>Random luck/accident in wealth attainment.</dd>
        <dt>Gene Homophily</dt>
        <dd>Assortative mating by genes (0 = random; 1 = nearest neighbors).</dd>
        <dt>Environment Homophily</dt>
        <dd>Assortative mating by environment (0 = random; 1 = nearest neighbors).</dd>
      </dl>
    `
  },
  {
    id: 'context',
    title: 'Political & Philosophical Context',
    content: `
      <h3>Political & Philosophical Context</h3>
      <p><strong>Gene vs. Environment slider extremes:</strong></p>
      <ul>
        <li><em>1 (genes only):</em> Utopian meritocracy where success depends solely on talent.</li>
        <li><em>0 (environment only):</em> Environmental determinism akin to nepotism or inherited privilege.</li>
      </ul>
      <p><strong>Education vs. Wealth slider extremes:</strong></p>
      <ul>
        <li><em>1 (education only):</em> Pure meritocracy (education drives wealth).</li>
        <li><em>0 (inheritance only):</em> Aristocracy (wealth inherited unchanged).</li>
      </ul>
      <p><strong>Combined extremes:</strong></p>
      <ul>
        <li>Full meritocracy: Gene→Education=1 & Education→Wealth=1.</li>
        <li>Full aristocracy: Gene→Education=0 & Education→Wealth=0.</li>
      </ul>
      <p><strong>Homophily sliders:</strong></p>
      <ul>
        <li>0 = random mating (no assortative pairing).</li>
        <li>1 = strict nearest‐neighbor mating (social segregation).</li>
      </ul>
      <p>Use these settings to explore thought experiments on equality of opportunity, inherited privilege, and social stratification.</p>
    `
  },
  {
    id: 'extensions',
    title: 'Extensions & Ideas',
    content: `
      <h3>Extensions & Experimental Interventions</h3>
      <ul>
        <li><strong>Targeted Supports:</strong> e.g. zero out Env Quality Noise for the bottom 20% to model educational aid for disadvantaged groups.</li>
        <li><strong>Genetic Interventions:</strong> e.g. override gene influences for low-scoring agents to simulate special education or medical treatments.</li>
        <li><strong>Redistribution Schemes:</strong> apply progressive/regressive taxes to final wealth and observe Gini changes.</li>
        <li><strong>Limitations:</strong> focuses on wealth, not broader quality of life or equality of condition (Sandel’s critique).</li>
      </ul>
    `
  },
  {
    id: 'extras',
    title: 'Extras & Tooltips',
    content: `
      <h3>Extras & Tooltips</h3>
      <ul>
        <li><strong>Hover</strong> over plots for precise values.</li>
        <li><strong>Legends</strong> map colors to numeric ranges.</li>
        <li><code>i</code> icon opens inheritance & inequality primer.</li>
        <li>Auto-update toggle re-runs the simulation live when sliders move.</li>
      </ul>
    `
  }
];
